rpc/block.go

```diff
@@ -278,6 +278,45 @@ func (h *Handler) BlockWithTxs(id BlockID) (*BlockWithTxs, *jsonrpc.Error) {
 	}, nil
 }
 
+type BlockWithTxsAndReceipts struct {
+    Status BlockStatus
+    BlockHeader
+    Transactions []*Transaction
+    Receipts []*TransactionReceipt
+}
+
+func (h *Handler) BlockWithTxsAndReceipts(id BlockID) (*BlockWithTxsAndReceipts, *jsonrpc.Error) {
+    block, rpcErr := h.blockByID(&id)
+    if rpcErr != nil {
+        return nil, rpcErr
+    }
+
+    blockStatus, rpcErr := h.blockStatus(id, block)
+    if rpcErr != nil {
+        return nil, rpcErr
+    }
+
+    finalityStatus := TxnAcceptedOnL2
+    if blockStatus == BlockAcceptedL1 {
+        finalityStatus = TxnAcceptedOnL1
+    }
+
+    txs := make([]*Transaction, len(block.Transactions))
+    receipts := make([]*TransactionReceipt, len(block.Transactions))
+    for index, txn := range block.Transactions {
```

Comment:
We implicitly assumed that `len(Transactions) == len(Receipts)`
in my opinion this deserves at least a comment.